businessRules/donations.js

```diff
@@ -21,34 +21,26 @@ function createDonation(firstName, sum, userId, ideaId) {
 }
 
 function ideasDonationSum(ideaId) {
-    const ideas = getAllIdeas();
-    const idea = ideas.find((idea) => idea.id === ideaId);
+    let idea = findIdeaWithTotalDonationSumWithSameId(ideaId);
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
-    }
     return idea.totalDonationSum;
 }
 
 function ideasSumDifference(ideaId) {
-    const ideas = getAllIdeas();
-    const idea = ideas.find((idea) => idea.id === ideaId);
-
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
-    }
+    let idea = checkIfIdeaIdMatchesIdeas(ideaId);
 
     const ideasSum = idea.askedSum;
 
     const totalDonationSumForThisIdea = ideasDonationSum(ideaId);
+
     const askedSumAndDonationSumDifference =
         ideasSum - totalDonationSumForThisIdea;
 
     return askedSumAndDonationSumDifference;
 }
 
 module.exports = {
-    createDonation,
+    createDonations,
```

Comment:
Why multiple? Does the function create multiple donations?

Reply:
Renamed.